src/main/java/org/jabref/gui/externalfiles/FindFullTextAction.java

```diff
@@ -70,14 +72,20 @@ public void execute() {
             }
         }
 
-        Task<Map<BibEntry, Optional<URL>>> findFullTextsTask = new Task<Map<BibEntry, Optional<URL>>>() {
+       Task<Map<BibEntry, Optional<URL>>> findFullTextsTask = new Task<Map<BibEntry, Optional<URL>>>() {
             @Override
             protected Map<BibEntry, Optional<URL>> call() {
                 Map<BibEntry, Optional<URL>> downloads = new ConcurrentHashMap<>();
                 int count = 0;
+                AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(basePanel.getBibDatabaseContext(), Globals.prefs.getFilePreferences(), Globals.prefs.getAutoLinkPreferences(), ExternalFileTypes.getInstance());
+                util.linkAssociatedFiles(basePanel.getSelectedEntries(), new NamedCompound(""));
+
                 for (BibEntry entry : basePanel.getSelectedEntries()) {
-                    FulltextFetchers fetchers = new FulltextFetchers(Globals.prefs.getImportFormatPreferences());
-                    downloads.put(entry, fetchers.findFullTextPDF(entry));
+                    List<LinkedFile> l = entry.getFiles();
```

Comment:
Please no single character or other abbreviation. Makes it harder for others to understand

Reply:
Ok, sorry. It's my first contribution, i will change.